discovery-provider/src/queries/queries.py

```diff
@@ -1872,3 +1872,146 @@ def get_top_followee_saves(type):
                     track['user'] = user
 
     return api_helpers.success_response(tracks)
+
+# Retrieves the top users for a requested genre under the follow parameters
+# - A given user can only be associated w/ one genre
+# - The user's associated genre is calculated by tallying the genre of the tracks and taking the max 
+#   - If there is a tie for # of tracks in a genre, then the first genre alphabetically is taken 
+# - The users associated w/ the requested genre are then sorted by follower count
+# Route Parameters
+#   urlParam: {Array<string>?}  genre       List of genres to query for the 'top' users
+#   urlParam: {boolean?}        with_user   Boolean if the response should be the user ID or user metadata defaults to false
+@bp.route("/top_genre_users", methods=("GET",))
+def get_top_genre_users():
+
+    genres = []
+    if "genre" in request.args:
+        genres = request.args.getlist("genre")
+
+    # If the with_users url arg is provided, then populate the user metadata else return user ids
+    with_users = "with_users" in request.args and request.args.get("with_users") != 'false'
+
+    db = get_db_read_replica()
+    with db.scoped_session() as session:
+        if (len(genres) == 0):
```

Comment:
@jowlee do we need another separate query if the genre is not passed in? could we just not use a conditional filter if a genre is passed in to select the appropriate genre?

Reply:
So we definitely don't need a separate query - We would modify the other query to not filter by genre & not filter tracks w/o a genre (Although the UI enforces a genre selection this should be added for safety b/c it is nullable in the DB). 
But, I believe the tradeoff is that the query is less performant. But yeah maybe it's worth it for the sake of simplicity and ease of understanding this code to have one code path here.